transactions/multi-document-acid-transactions-mongodb-go.md

```diff
@@ -0,0 +1,214 @@
+# Multi-Document ACID Transactions in MongoDB with Go
+
+The past few months have been an adventure when it comes to getting started with MongoDB using the Go programming language (Golang). We've explored everything from create, retrieve, update, and delete (CRUD) operations, to data modeling, and to change streams. To bring this series to a solid finish, we're going to take a look at a popular requirement that a lot of organizations need, and that requirement is transactions.
+
+So why would you want transactions?
+
+There are some situations where you might need atomicity of reads and writes to multiple documents within a single collection or multiple collections. This isn't always a necessity, but in some cases it might be.
+
+Take the following for example.
+
+Let's say you want to create documents in one collection that depend on documents in another collection existing. Or let's say you have schema validation rules in place on your collection. In the scenario that you're trying to create documents and the related document doesn't exist or your schema validation rules fail, you don't want the operation to proceed. Instead, you'd probably want to rollback to before it happened.
+
+There are other reasons that you might use transactions, but you can use your imagination for those.
+
+In this tutorial, we're going to look at what it takes to use transactions with Golang and MongoDB. Our example will rely more on schema validation rules passing, but it isn't a limitation.
+
+## Understanding the Data Model and Applying Schema Validation
```

Comment:
This section is a little confusing because you talk about schema validation, but I don't see anything about actually creating the collection with the validator. You can either show them the command they can run via the mongo shell, or use `Database.RunCommand` to create the collection in the linked code before doing anything with sessions/transactions.

Reply:
I had mentioned this:

"The above logic would be applied using the MongoDB CLI or with Compass ..."

I also link to another tutorial focused strictly on validation.

Do you think that is not enough? My goal was to not turn this into a validation blog post which is why I went brief on the subject.

Thoughts?

Reply:
I missed it but I was more skimming and looking for issues in the code/post so maybe a more attentive reader who's actually reading this as a tutorial would see it. I'll defer to your judgement here.